x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/esnative/UserAndPassword.java

```diff
@@ -34,6 +36,16 @@ public User user() {
         return this.passwordHash;
     }
 
+    boolean verifyPassword(SecureString data) {
+        try {
+            Hasher hasher = Hasher.resolveFromHash(this.passwordHash);
+            return hasher.verify(data, this.passwordHash);
+        } catch (IllegalArgumentException e) {
+            // The hash was of invalid format, we can't validate the password
+            return false;
+        }
```

Comment:
I think `hasher`can be resolved in the constructor and throwing `IllegalArgumentException` theere is fine.